nodeup/pkg/model/etcd_tls.go

```diff
@@ -22,19 +22,19 @@ import (
 	"k8s.io/kops/upup/pkg/fi"
 )
 
-// CalicoBuilder configures the calico CNI provider
-type CalicoBuilder struct {
+// EtcdTLSBuilder configures the etcd TLS support
+type EtcdTLSBuilder struct {
 	*NodeupModelContext
 }
 
-var _ fi.ModelBuilder = &CalicoBuilder{}
+var _ fi.ModelBuilder = &EtcdTLSBuilder{}
 
-// Build is responsible for performing any setup to the calico CNI provider
-func (b *CalicoBuilder) Build(c *fi.ModelBuilderContext) error {
+// Build is responsible for performing setup for CNIs that need etcd TLS support
+func (b *EtcdTLSBuilder) Build(c *fi.ModelBuilderContext) error {
 	// @check if tls is enabled and if so, we need to download the client certificates
 	if b.UseEtcdTLS() {
-		name := "calico-client"
-		dirname := "calico"
+		name := "etcd-client"
+		dirname := "etcd-secrets"
```

Comment:
hmm .. in theory, although we don't it now, by using separate client certificate we still have the options of locking users down via roles in etcd rather than an allow all ... Not my call though @chrislovecnm @justinsb

Reply:
This wasn't an issue when merging calico tls support, I assumed it would be fine in generalized way too.

Reply:
In general nobody should be talking to etcd; everyone should be talking to apiserver instead (e.g. using a CRD, or using the CIDR or another on the node).  Flannel deprecated it, Calico is working on it.  I don't know if you're a Cilium developer @nebril , but hopefully there's a long term plan to get away from etcd access.

(That said, aggregated api servers do talk directly to etcd, so there's an ongoing discussion about how those should work - should they run their own etcd or should they share an etcd, e.g. via a proxy)

Reply:
(And specifically here I think we can lock these more down in future)